correct parsing values for slice of strings with commas

sattellite opened the pull request:
Slice of strings can contain strings with commas. Fixed the function for represent such slice into a string. The special unicode character is used as separator, which is described as "record separator".

Fix for #160

sattellite committed:
fix of parsing slice of strings with commas

sattellite committed:
fixed mistyping in comment

sattellite commented:
@cristaloleg help to review the PR

cristaloleg commented:
Something isn't ok https://github.com/cristalhq/aconfig/actions/runs/14853854779/job/42111518539?pr=161#step:12:113

cristaloleg commented:
I think https://github.com/cristalhq/aconfig/issues/63 should be reopened and the proper solution should be: everyone can set it's own slice separator (which should be default to `,`).

sattellite commented:
thanks, i will look for solution.

sattellite committed:
make the slice separator customizable

sattellite commented:
@cristaloleg i have changed approach. added the SliceSeparator option, which is passed to Loader so that it can use it to work with slices. all tests are passed

sattellite commented on utils.go:
```diff
@@ -179,13 +179,13 @@ func (d *jsonDecoder) DecodeFile(filename string) (map[string]interface{}, error
 }
 
-func sliceToString(curr interface{}) string {
+func (l *Loader) sliceToString(curr interface{}) string {
```
may be move it to reflection.go?

sattellite commented:
could you please create a new release?
